### PR TITLE
autoplay ios

### PR DESCRIPTION
**Pull request by emmyrlim**
trying out toggling play from RN

obligatory "first"

### CODE REVIEW

File: ios/TransparentVideoViewManager.swift [hunk 1]
```diff
@@ -103,14 +110,18 @@ class TransparentVideoView : UIView {
         case .failure(let error):
           return print("Something went wrong when loading our video", error)
 
-        case .success(let player):
+        case .success(let player) where self?.videoAutoplay == true:
           // Finally, we can start playing
           player.play()
+
+        case .success(let player):
+          // Finally, we can start playing
```
Comment (pmick) [inline ios/TransparentVideoViewManager.swift:118]:
```suggestion
```
I think we can 🚮

File: example/src/components/ParallaxVideo.tsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ const ParallaxVideo = ({ source, zIndex }: any) => {
 
   return (
     <Animated.View style={[styles.container, layerStyle]}>
-      <TransparentVideo source={source} style={styles.video} />
+      <TransparentVideo source={source} style={styles.video} autoplay />
```
Comment (pmick) [inline example/src/components/ParallaxVideo.tsx:47]: My knowledge of tsx is somewhat limited. In this context, where is `autoplay` coming from?

File: ios/TransparentVideoViewManager.swift [hunk 3]
```diff
@@ -17,20 +17,27 @@ class TransparentVideoView : UIView {
...
+    }
+  }
+
+  @objc var autoplay: Bool = false {
+    didSet {
+      self.videoAutoplay = autoplay
       let itemUrl = URL(string: self.source!.uri!)!
       loadVideoPlayer(itemUrl: itemUrl)
     }
   }
```
Comment (pmick) [inline ios/TransparentVideoViewManager.swift:34]: I'm not sure what order react native will set these variables in. I wonder if it's based on the order that the view props are set in the `TransparentVideoViewManager.m` file 🤔 . 

I worry if it sets `autoplay` first, with the force unwraps on line 31 we'd crash. I wonder if we should call loadVideoPlayer in both of these `didSet` scopes and early return in `loadVideoPlayer` if the url on the player is equal to the url we're trying to set. Basically make it do nothing if it gets called with the same url many times.

File: example/src/components/ParallaxVideo.tsx (hunk 2 above)
Comment (emmyrlim) [reply example/src/components/ParallaxVideo.tsx:47]: ah, this is only the example. But basically in this case, putting autoplay here is essentially the equivalent of saying `autoplay={true}` (as compared to `autoplay={autoplay}`

File: ios/TransparentVideoViewManager.swift (hunk 3 above)
Comment (emmyrlim) [reply ios/TransparentVideoViewManager.swift:34]: oh that's pretty good, lemme try that.

**Review by TimWatervoort — APPROVED** [review]
LGTM, but fair warning, I've only ever written Obj-C, not Swift 😬 
May want a second opinion

**Pull request merged**